includes/reader-activation/class-reader-activation.php

```diff
@@ -1514,32 +1515,37 @@ public static function rate_limit_lost_password( $errors, $user_data ) {
 
 	/**
 	 * If a reader tries to make a recurring donation with an email address that
-	 * has been previously registered, force Woo to allow the subscription to
-	 * be created and associated with the existing account without requiring the
-	 * reader to log in.
+	 * has been previously registered, automatically associate the transaction with the user.
+	 *
+	 * @param int $customer_id Current customer ID.
+	 * @return int Modified $customer_id
 	 */
-	public static function allow_subscription_purchase_without_login() {
-		if ( ! class_exists( '\WC_Checkout' ) ) {
-			return;
+	public static function associate_existing_woo_users_with_transactions_on_checkout( $customer_id ) {
+		$billing_email = filter_input( INPUT_POST, 'billing_email', FILTER_SANITIZE_EMAIL );
+		if ( $billing_email ) {
+			$customer = \get_user_by( 'email', $billing_email );
+			if ( $customer ) {
+				$customer_id = $customer->ID;
+			}
 		}
+		return $customer_id;
+	}
 
-		$wc_checkout = new \WC_Checkout();
-		$posted_data = $wc_checkout->get_posted_data();
-
-		if ( empty( $posted_data ) || empty( $posted_data['billing_email'] || empty( $posted_data['createaccount'] ) ) ) {
-			return;
+	/**
+	 * Don't force account registration/login on Woo purchases for existing users.
+	 *
+	 * @param array $data Array of Woo checkout data.
+	 * @return array Modified $data.
+	 */
+	public static function dont_force_registration_for_existing_woo_users( $data ) {
+		$email    = $data['billing_email'];
+		$customer = \get_user_by( 'email', $email );
+		if ( $customer ) {
+			$data['createaccount'] = 0;
+			\add_filter( 'woocommerce_checkout_registration_required', '__return_false', 99 );
```

Comment:
I'm not sure if `9999` is required, but it didn't work with `999`, so I went up another digit.

```suggestion
			\add_filter( 'woocommerce_checkout_registration_required', '__return_false', 9999 );
```